activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -286,6 +286,10 @@ def add_to_target(record, skip_callbacks = false, &block)
         replace_on_target(record, index, skip_callbacks, &block)
       end
 
+      def target=(record)
+        @target = [record].flatten(1)
```

Comment:
ActiveStorage attachments use `delegate_missing_to` to delegate to blobs to see if the object responds to `#to_ary` if I use `Array(record)` which then loads the association and a looping assignment of `target=`. I'm open to a better alternative, but I think this is acceptable if we can assume we don't want to ask record any questions.